### PR TITLE
JSX parent child Assignment

### PR DESCRIPTION
**Pull request by aryaZemo**
Kindly Review my ParentChild Assignment

### CODE REVIEW

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx:1]: Change this file name also !

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { Story, Meta } from "@storybook/react";
+
+// import TextBox, { TextBoxProps } from './TextBox';
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx:4]: remove the commented out code if not needed

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import TextBox from '../../atmos/CustomTypographyTextBox/CustomTypographyTextBox'
+import { ReactComponent as IButtonIcon } from "./info-circle.svg";
+import { DueDateChildComponent } from './DueDateChildComponent'
+
+const StyledDate = styled.div`
+ 
+  display: flex;
+  gap: 8px;
+  width: 174px;
+  height: 22px;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:11]: Don`t hardcode height and width instead give them in vw( for width ) and vh( for height )

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  padding: 4px 8px;
+  gap: 10px;
+  position: absolute;
+  width: 123px;
+  height: 22px;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:30]: Don`t hardcode height and width instead give them in vw( for width ) and vh( for height )

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  };
+
+  const formattedDueDate = formatDate(dueDate);
+  
+  return (
+    <>
+      <StyledDate>
+        
+          <TextBox 
+            colors="#C9C8CC"
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:69]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+display: flex;
+flex-direction: row;
+justify-content: center;
+align-items: center;
+padding: 20px;
+gap: 20px;
+width: 80px;
+height: 80px;
+background: #262529;
+border: 1px solid #28272B;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx:17]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  top: 20px;
+  background: #e39ab2;
+  border-radius: 4px;
+  flex: none;
+  order: 1;
+  flex-grow: 0;
+  z-index: 1;
+`;
+const TextColor = styled.div`
+  color: #201f24;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:41]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  justify-content: center;
+  align-items: center;
+  padding: 4px 8px;
+  gap: 10px;
+  position: absolute;
+  width: 123px;
+  height: 22px;
+  right: 20px;
+  top: 20px;
+  background: #e39ab2;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:33]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+box-sizing: border-box;
+display: flex;
+flex-direction: row;
+justify-content: center;
+align-items: center;
+padding: 20px;
+gap: 20px;
+width: 80px;
+height: 80px;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx:15]: Don`t hardcode height and width instead give them in vw( for width ) and vh( for height )

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  box-sizing: border-box;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 32px;
+  gap: 12px;
+  isolation: isolate;
+  position: absolute;
+  width: 340px;
+  height: 259px;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:17]: Don`t hardcode height and width instead give them in vw( for width ) and vh( for height )

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  left: 278px;
+  top: 143px;
+  background: #201f24;
+  border: 1px solid #28272b;
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:21]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+align-items: flex-start;
+`
+
+const CardComponent = () => {
+  return (
+    <StyledCard>
+      <IconCard></IconCard>
+      <DateComponent></DateComponent>
+      <StyledDollar>
+        <TextBox colors='#E8E7F0' variation='h2' text='$14,204.55'></TextBox>
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:38]: Don`t hardcode the color codes, instead get them from theme file

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/receipt-text.svg
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/receipt-text.svg:1]: move this public folder you can maintain a image folder in public folder

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/info-circle.svg
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/info-circle.svg:1]: move this public folder you can maintain a image folder in public folder

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+flex: none;
+order: 0;
+flex-grow: 0;
+margin-top: 20.5px;
+`
+
+const IconCard = () => {
+  return (
+    <StyledIconCard>
+      <Icon></Icon>
```
Comment (Sahithi-003) [inline React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx:28]: where are you passing the image path here?

Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx:28]: I am importing the Icon as a component  I have that icon in the form of .svg format. then in this file, I am doing ...

import {ReactComponent as Icon} from './receipt-text.svg'

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/info-circle.svg
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/info-circle.svg:1]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/receipt-text.svg
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/receipt-text.svg:1]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx (hunk 11 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:38]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx (hunk 10 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:21]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx (hunk 9 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/organism/Card/Card.tsx:17]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx (hunk 8 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/IconCard/IconCard.tsx:15]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx (hunk 7 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:33]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx (hunk 6 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:41]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx (hunk 4 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:69]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx (hunk 1 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/atmos/CustomTypographyTextBox/TextBox.stories.tsx:4]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx (hunk 2 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:11]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx (hunk 3 above)
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/Date.tsx:30]: done

File: React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/DueDateChildComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,10 @@
+interface Props {
+    daysRemaining: number;
+  }
+  
+  export const DueDateChildComponent = ({ daysRemaining }: Props) => {
+    return (
+      <p>Due in {daysRemaining} {daysRemaining === 1 ? "day" : "day(s)"}</p>
```
[the remark this answers is not in the document]
Comment (aryaZemo) [reply React/JSX-ParentChild/jsx-parent-child-assignment/src/Assignment/molecules/Date/DueDateChildComponent.tsx:7]: done

**Review by Sahithi-003 — APPROVED** [review]
Remove commented code if not needed

**Pull request merged**